site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx

```diff
@@ -220,11 +220,49 @@ const CreateConsequenceServices = (props: CreateConsequenceServicesProps): React
             });
         } else {
             if (stopToAdd) {
+                const servicesForGivenStop = await fetchServicesByStops({
+                    atcoCodes: [stopToAdd.atcoCode],
+                    includeRoutes: true,
+                    dataSource: dataSource,
+                });
```

Comment:
When multiple services are connected to a stop, when I add a stop then delete the service that I originally searched for the stop attached gets removed but the other services remain - is that what we expect?
For example:
Searched for 1 - Barnsley Interchange - Staincross Circular (Stagecoach)
Selected 1 - Barnsley Interchange - Staincross Circular (Stagecoach)
Searched Bar Lane/Cloudberry Way adj 370050414
Selected Bar Lane/Cloudberry Way adj 370050414
Delete 1 - Barnsley Interchange - Staincross Circular (Stagecoach)
Bar Lane/Cloudberry Way adj 370050414 gets deleted but the 486 - Darton College - Mapplewell (Globe Holidays) and 488 - Darton Academy - Monk Bretton (Globe Holidays) remain

Reply:
Updated dropdown logic so that a stop stays selected until all services that include that stops are removed.

Also updated the logic on the stop dropdown options to ensure only options for selected services are shown. 

(Previously, if you selected a stop that added some additional services the stop dropdown options would be updated to include these additional services. However, if you deleted one of those additional services their stop options would sometimes remain in the stop options dropdown)

Reply:
The select all button for Barnsley Interchange - Staincross Circular (Stagecoach) does not select all related services to every stop

Reply:
Chris confirmed that when clicking the select all button we should only go and grab all the stops related to the selected services. We shouldn't select all the related services to all the stops.